NumLib/Assembler/VectorMatrixAssembler.h

```diff
@@ -95,17 +95,14 @@ class VectorMatrixAssembler<GlobalMatrix, GlobalVector, LocalAssembler,
         const double t, GlobalVector const& x,
         GlobalMatrix& M, GlobalMatrix& K, GlobalVector& b) const
     {
-        auto cb = [&local_assembler](
-                std::vector<double> const& local_x,
-                LocalToGlobalIndexMap::RowColumnIndices const& r_c_indices,
-                const double t,
-                GlobalMatrix& M, GlobalMatrix& K, GlobalVector& b)
-        {
-            local_assembler.assemble(t, local_x);
-            local_assembler.addToGlobal(r_c_indices, M, K, b);
-        };
-
-        passLocalVector_(cb, id, _data_pos, x, t, M, K, b);
+        auto const x_and_indices = prepareLocalVector(id, _data_pos, x);
```

Comment:
still i don't get it. why can't you just pass an indices vector to prepareLocalVector()? then the query is made only once. And I also don't get when the local DOFs and the corresponding indices don't match.